notion/src/components/sidebar/SideNavbar.js

```diff
@@ -0,0 +1,48 @@
+import { deleteDocument, getDocument, postDocument } from "../../api/api.js";
+import { push } from "../../route/router.js";
+import DocumentList from "./DocumentList.js";
+import LinkButton from "../linkbutton/LinkButton.js";
+import SidebarHeader from "./SidebarHeader.js";
+
+export default function SideNavbar({ parent, initialState }) {
+  const page = document.createElement('div');
+  page.id = 'documents-page';
+  
+  new SidebarHeader({ 
+    parent: page,
+    closeSideNavbar: () => {
+      page.style = 'transform: translateX(-281px)';
+    },
+    openSideNavbar: () => {
+      page.style = 'transform: translateX(0px)';
+    }
+  })
+
+  const documentList = new DocumentList({
+    parent: page,
+    initialState,
+  })
+
+  new LinkButton({
+    $target: page,
+    initialState: {
+      text: '+ 페이지 추가',
+      id: 'add-document-button'
+    },
+    action: async () => {
+      const { id } = await postDocument('/documents');
+      push(`/documents/${id}`);
+      this.setState();
+    }
+  })
+  
+  this.setState = async () => {
+    const documents = await getDocument('/documents');
+
+    documentList.setState(documents);
+  }
```

Comment:
```suggestion
this.setState = async () => {
    const documents = await getDocument("/documents");
    documentList.setState(documents);
    this.render();
  };
```
일단 이렇게하면 리렌더링이 좀 거슬리긴해도 작동은 하는것 같습니다!! 아마 렌더 코드를 넣으시려다가 바쁘셔서 중간에 놓치신것 같은데요...ㅎㅎ